enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,248 @@
+Some bugs only impact specific platforms, or clusters with other specific features.
+For example, rhbz#1858026 [only impacted][rhbz-1858026-impact-statement] clusters with the `None` platform which were created as 4.1 clusters and subsequently updated via 4.2, 4.3, and 4.4 to reach 4.5.
+In those cases there is currently tension between wanting to protect vulnerable clusters by blocking the edge vs. wanting to avoid inconveniencing clusters which we know are not vulnerable and whose administrators may have been planning on taking the recommended update.
+This enhancement aims to reduce that tension.
+
+### Goals
+
+* [Cincinnati graph-data][graph-data] maintainers will have the ability to block edges for the subset of clusters matching a particular Prometheus query.
+
+### Non-Goals
+
+* Blocking edges based on data that is not [uploaded to Telemetry][uploaded-telemetry].
+* Blocking edges for subsets of clusters when the update service is unable to reach or authenticate with an aggregating Prometheus server.
+    For example, users running a local update service will not be able to access Red Hat's internal Telemetry to determine the requesting cluster's platform, etc.
+* Exactly scoping the set of blocked clusters to those which would have been impacted by the issue.
+    For example, some issues may be races where the impacted cluster set is a random subset of the vulnerable cluster set.
+    Any targeting of the blocked edges will reduce the number of blocked clusters which would have not been impacted, and thus reduce the contention between protecting vulnerable clusters and inconveniencing invulnerable clusters.
+* Specifying a particular update service implementation.
+    This enhancement floats some ideas, but the details of the chosen approach are up to each update service's maintainers.
+
+## Proposal
+
+### Enhanced graph-data schema for blocking edges
+
+[The blocked-edges schema][block-edges] will be extended with the following new properties:
+
+* `clusters` (optional, [object][json-object]), defining the subset of affected clusters.
+    If any `clusters` property matches a given cluster, the edge should be blocked for that cluster.
+    * `promql` (optional, [object][json-string]), with a [PromQL][] query describing affected clusters.
+        This query will be submitted to a configurable Prometheus service and should return a set of matching records with `_id` labels.
+        Clusters whose [submitted `id` query parameter][cincinnati-for-openshift-request] is in the set of returned IDs are allowed if the value of the matching record is 1 and blocked if the value of the matching record is 0.
+        Clusters whose submitted `id` is not in the result set, or which provide no `id` parameter, are also blocked.
+        Blocking too many edges is better than blocking too few, because you can recover from the former, but possibly brick clusters on the latter.
```

Comment:
I do not think this is a good idea. IMO we should do the opposite i.e. we should not block the cluster if the query does not provide no `id` parameter. Because I am afraid that we will block clusters and we wont be knowing which clusters are blocked. When we block a cluster we should which ones are blocked so that we can unblock them in future. CEE can reach out to them if required.